contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,221 @@
+            revert Errors.InsufficientBuybackOutputAmount(feeToken, buybackToken, outputAmount, minOutputAmount);
+        }
 
+        _buybackFees(feeToken, buybackToken, batchSize, outputAmount, availableFeeAmount);
+    }
+
+    function getOutputAmount(
+        address[] calldata markets,
+        address feeToken,
+        address buybackToken,
+        uint256 version
+    ) external view returns (uint256) {
+        uint256 batchSize = _getBatchSize(buybackToken);
+        _validateBuybackToken(batchSize, buybackToken);
+
+        uint256 feeAmount;
+        uint256 availableFeeAmount = _getAvailableFeeAmount(feeToken, buybackToken);
         for (uint256 i; i < markets.length; i++) {
-            FeeUtils.claimFees(
-                dataStore,
-                eventEmitter,
-                markets[i],
-                tokens[i],
-                receiver
-            );
+            if (version == v1) {
+                feeAmount = vaultV1.feeReserves(feeToken);
+            } else if (version == v2) {
+                feeAmount = _getUint(Keys.claimableFeeAmountKey(markets[i], feeToken));
+            } else {
+                revert Errors.InvalidVersion(version);
+            }
+
+            FeeAmounts memory feeAmounts = _getFeeAmounts(version, feeAmount);
+            feeAmount = buybackToken == gmx ? feeAmounts.gmx : feeAmounts.wnt;
+            availableFeeAmount = availableFeeAmount + feeAmount;
+        }
+
+        uint256 maxFeeTokenAmount = _getMaxFeeTokenAmount(feeToken, buybackToken, batchSize);
+        if (availableFeeAmount >= maxFeeTokenAmount) {
+            return maxFeeTokenAmount;
+        } else {
+            return availableFeeAmount;
+        }
+    }
+
+    function _buybackFees(
+        address feeToken,
+        address buybackToken,
+        uint256 batchSize,
+        uint256 buybackAmount,
+        uint256 availableFeeAmount
+    ) private {
+        IERC20(buybackToken).safeTransferFrom(msg.sender, address(this), batchSize);
+        IERC20(feeToken).safeTransfer(msg.sender, buybackAmount);
+
+        _setAvailableFeeAmount(feeToken, buybackToken, availableFeeAmount - buybackAmount);
+    }
+
+    function _incrementAvailableFeeAmounts(uint256 version, address feeToken, uint256 feeAmount) private {
+        address wnt = dataStore.getAddress(Keys.WNT);
+
+        FeeAmounts memory feeAmounts = _getFeeAmounts(version, feeAmount);
+
+        uint256 availableFeeAmountGmx = _getAvailableFeeAmount(feeToken, gmx) + feeAmounts.gmx;
+        uint256 availableFeeAmountWnt = _getAvailableFeeAmount(feeToken, wnt) + feeAmounts.wnt;
+
+        _setAvailableFeeAmount(feeToken, gmx, availableFeeAmountGmx);
+        _setAvailableFeeAmount(feeToken, wnt, availableFeeAmountWnt);
+    }
+
+    function _setAvailableFeeAmount(address feeToken, address buybackToken, uint256 availableFeeAmount) private {
+        dataStore.setUint(Keys.buybackAvailableFeeAmountKey(feeToken, buybackToken), availableFeeAmount);
```

Comment:
should emit a SetAvailableFeeAmount event here

Reply:
Ok, this has been added in the latest commit, not sure if buybackToken (and maybe feeToken) should be indexed?